src/internal.jl

```diff
@@ -0,0 +1,20 @@
+
+import UUIDs
+
+"""Internal storage common to PowerSystems types."""
+struct PowerSystemInternal
+    uuid::Base.UUID
+end
+
+"""Creates PowerSystemInternal with a UUID."""
+PowerSystemInternal() = PowerSystemInternal(UUIDs.uuid4())
+
+"""Gets the UUID for any PowerSystemType."""
+function get_uuid(obj::T)::Base.UUID where T <: PowerSystemType
+    return obj.internal.uuid
+end
+
+"""Gets the UUIDs for a vector of any PowerSystemType."""
+function get_uuids(objs::Vector{T})::Vector{Base.UUID} where T <: PowerSystemType
```

Comment:
I have a minor preference for renaming this to `get_uuid`. 

I think I also prefer just removing this function. `get_uuid.(objs, uuids)` will call the previous method and return `Vector{Base.UUID}` appropriately because of the broadcasting operator.

Reply:
Thanks!  I forgot about the broadcast operator.  Deleted.